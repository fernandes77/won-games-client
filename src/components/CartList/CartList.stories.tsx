import type { Meta, StoryObj } from '@storybook/react'
import { CartList } from './CartList'
import items from './CartList.mock'

const meta: Meta<typeof CartList> = {
  component: CartList,
  decorators: [
    (Story) => (
      <div style={{ maxWidth: 800 }}>
        <Story />
      </div>
    )
  ]
}

export default meta
type Story = StoryObj<typeof CartList>

export const Primary: Story = {
  args: { items, total: 'R$ 430,00' }
}

export const WithButton: Story = {
  args: { items, total: 'R$ 430,00', hasButton: true }
}

export const Empty: Story = {
  decorators: [
    (Story) => (
      <div style={{ maxWidth: 800 }}>
        <CartList />
      </div>
    )
  ]
}
