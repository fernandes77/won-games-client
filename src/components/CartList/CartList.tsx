import { GameItem, GameItemProps } from '@/components/GameItem/GameItem'

export type CartListProps = {
  items: GameItemProps[]
  total: string
}

export const CartList = ({ items, total }: CartListProps) => (
  <main className="bg-white flex flex-col self-start">
    {items.map((item) => (
      <GameItem key={item.title} {...item} />
    ))}

    <div
      className="bg-gray-lighter text-black font-bold text-sm p-5 flex justify-between
      min-md:text-md min-md:p-sm"
    >
      Total <span className="text-primary">{total}</span>
    </div>
  </main>
)
