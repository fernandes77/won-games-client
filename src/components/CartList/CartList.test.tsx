import { render, screen } from '@/utils/tests/helpers'
import { CartList } from './CartList'
import mockItems from './CartList.mock'

describe('<CartList />', () => {
  it('should render the cart list', () => {
    const { container } = render(
      <CartList items={mockItems} total="R$ 430,00" />
    )

    expect(screen.getAllByRole('heading')).toHaveLength(2)
    expect(screen.getByText('R$ 430,00')).toHaveClass('text-primary')

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button', () => {
    render(<CartList items={mockItems} total="R$ 430,00" hasButton />)

    expect(screen.getByText(/buy it now/i)).toBeInTheDocument()
  })
})
