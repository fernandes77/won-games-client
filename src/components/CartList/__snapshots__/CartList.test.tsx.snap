// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartList /> should render the cart list 1`] = `
<main
  class="bg-white flex flex-col self-start"
>
  <div
    class="p-sm border-b border-gray-light min-md:flex"
  >
    <div
      class="flex"
    >
      <div
        class="shrink-0 mr-3 w-24 h-14 min-md:mr-xs min-md:w-[9.4rem] min-md:h-[4.4rem]"
      >
        <img
          alt="Red Dead Redemption 2"
          class="w-full h-full object-cover"
          src="https://source.unsplash.com/user/willianjusten/151x70"
        />
      </div>
      <div
        class="flex flex-col items-start"
      >
        <h3
          class="flex items-center text-sm text-black mb-xxs min-md:text-xl"
        >
          Red Dead Redemption 2
        </h3>
        <div
          class="text-white py-[2px] px-xxs bg-secondary rounded text-sm font-bold"
        >
          R$ 215,00
        </div>
      </div>
    </div>
  </div>
  <div
    class="p-sm border-b border-gray-light min-md:flex"
  >
    <div
      class="flex"
    >
      <div
        class="shrink-0 mr-3 w-24 h-14 min-md:mr-xs min-md:w-[9.4rem] min-md:h-[4.4rem]"
      >
        <img
          alt="Borderlands 3"
          class="w-full h-full object-cover"
          src="https://source.unsplash.com/user/willianjusten/151x70"
        />
      </div>
      <div
        class="flex flex-col items-start"
      >
        <h3
          class="flex items-center text-sm text-black mb-xxs min-md:text-xl"
        >
          Borderlands 3
        </h3>
        <div
          class="text-white py-[2px] px-xxs bg-secondary rounded text-sm font-bold"
        >
          R$ 215,00
        </div>
      </div>
    </div>
  </div>
  <div
    class="bg-gray-lighter text-black font-bold text-sm p-5 flex justify-between min-md:text-md min-md:p-sm"
  >
    Total 
    <span
      class="text-primary"
    >
      R$ 430,00
    </span>
  </div>
</main>
`;
